test/threadsafe_function/threadsafe_function.cc

```diff
@@ -0,0 +1,185 @@
+#include <uv.h>
```

Comment:
I think we want a big comment here that says this using uv is not guaranteed to be ABI stable.  That if you want to use the stability guarantee from N-API you need to use other means to implement threads.  This is the first example where we have used uv in the testing and we want it to be clear that it is not what we recommend and that we only do it because we know the test suite will be build/run with Node.js as opposed to being built into a binary that will run across different Node.js versions.

Reply:
Hmmm ... we could use `std::thread` instead, right? I think all our supported compilers now have C++11, right? In the test in core I used libuv because, well, we maintain core ourselves, and because in core I seek to keep N-API tests written in C as much as possible. @romandev could you convert this to `std::thread`?

Reply:
That would be better.